fanova/visualizer.py

```diff
@@ -167,23 +172,24 @@ def plot_pairwise_marginal(self, param_list, resolution=20, show=False, three_d=
                 plt.xlabel(param_names[0])
                 plt.ylabel(param_names[1])
                 plt.colorbar().set_label(self._y_label)
-                
+
                 if show:
                     plt.show()
             else:
-                cat_choices = self.cs_params[param_indices[0]].choices if isinstance(self.cs_params[param_indices[0]], (CategoricalHyperparameter)) else self.cs_params[param_indices[1]].choices
-                cats = choices[0] if (choices[0][0] in cat_choices and choices[0][-1] in cat_choices) else choices[1]
-                x_label = param_names[0] if isinstance(self.cs_params[param_list[1]],(CategoricalHyperparameter)) else param_names[1]
-                
+                # Only one of them is categorical
+                cat_choices = self.cs_params[param_indices[0]].choices if first_is_cat else self.cs_params[param_indices[1]].choices
+                cats = choices[0] if all(c in cat_choices for c in choices[0]) else choices[1]
+                x_label = param_names[0] if second_is_cat else param_names[1]
+
                 fig = plt.figure()
                 for i, cat in enumerate(cats):
```

Comment:
If you agree we would then have to also change the following
```suggestion
                for i, cat in enumerate(cat_choices):
```